dipy/reconst/dki.py

```diff
@@ -893,6 +950,158 @@ def axial_kurtosis(dki_params, min_kurtosis=0, max_kurtosis=3):
+    max_value = max(max_val)
+    max_direction = max_dir[np.argmax(max_val.argmax)]
+
+    # refine maxima direction
+    if gtol is not None:
+        for p in range(n):
+            r, theta, phi = cart2sphere(max_dir[p, 0], max_dir[p, 1],
+                                        max_dir[p, 2])
+            ang = np.array([theta, phi])
+            ang[:] = opt.fmin_bfgs(_kt_maxima_converge, ang, args=(dt, MD, kt),
+                                   gtol=gtol, disp=False, retall=False)
+            k_dir = np.array([sphere2cart(1., ang[0], ang[1])])
+            k_val = _directional_kurtosis(dt, MD, kt, k_dir)
+            if k_val > max_value:
+                max_value = k_val
+                max_direction = k_dir
+
+    return max_value, max_direction
+
+
+def kurtosis_maximum(dki_params, sphere='symmetric362', gtol=1e-5,
+                     mask=None):
+    """ Computes kurtosis maxima value
+
+    Parameters
+    ----------
+    dki_params : ndarray (x, y, z, 27) or (n, 27)
+        All parameters estimated from the diffusion kurtosis model.
+        Parameters are ordered as follow:
+            1) Three diffusion tensor's eingenvalues
+            2) Three lines of the eigenvector matrix each containing the first,
+               second and third coordinates of the eigenvector
+            3) Fifteen elements of the kurtosis tensor
+    sphere : Sphere class instance, optional
+        The sphere providing sample directions for the initial search of the
+        maximal value of kurtosis.
+    gtol : float, optional
+        This input is to refine kurtosis maxima under the precision of the
+        directions sampled on the sphere class instance. The gradient of the
+        convergence procedure must be less than gtol before successful
+        termination. If gtol is None, fiber direction is directly taken from
+        the initial sampled directions of the given sphere object
+    mask : ndarray
+        A boolean array used to mark the coordinates in the data that should be
+        analyzed that has the shape dki_params.shape[:-1]
+
+    Returns
+    --------
+    max_value : float
+        kurtosis tensor maxima value
+    max_dir : array (3,)
+        Cartesian coordinates of the direction of the maximal kurtosis value
+    """
+    shape = dki_params.shape[:-1]
+
+    # load gradient directions
+    if not isinstance(sphere, dps.Sphere):
+        sphere = get_sphere('symmetric362')
+
+    # select voxels where to find fiber directions
+    if mask is None:
+        mask = np.ones(shape, dtype='bool')
+    else:
+        if mask.shape != shape:
+            raise ValueError("Mask is not the same shape as dki_params.")
+
+    evals, evecs, kt = split_dki_param(dki_params)
+
+    # select non-zero voxels
+    pos_evals = _positive_evals(evals[..., 0], evals[..., 1], evals[..., 2])
+    mask = np.logical_and(mask, pos_evals)
+
+    kt_max = np.zeros(mask.shape)
+
+    for idx in ndindex(shape):
+        if not mask[idx]:
```

Comment:
This could be simplified as: 

    for idx in np.where(mask):

That way, you don't have to check whether the mask has a value in each iteration (and potentially loop over plenty of voxels you don't care about, doing that).